File: src/main/java/cz/skup5/jEvropa2/data/Item.java
package cz.skup5.jEvropa2.data;

import java.net.URL;

/**
 *
 * @author Roman
 */
public class Item extends E2Data {

    private URL webSiteUrl, imgUrl, mediaUrl;
    private String time;

    public Item(String name, URL imgUrl, String time) {
        this(name, EMPTY_URL, imgUrl, EMPTY_URL, time);
    }

    public Item(String name, URL imgUrl, URL mediaUrl, String time){
        this(name, EMPTY_URL, imgUrl, mediaUrl, time);
    }
   
    public Item(String name,  String time, URL webSiteUrl, URL imgUrl) {
        this(name, webSiteUrl, imgUrl, EMPTY_URL, time);
    }

    public Item(String name, URL webSiteUrl, URL imgUrl, URL mediaUrl, String time) {
        super(name);
        this.webSiteUrl = webSiteUrl;
        this.imgUrl = imgUrl;
        this.mediaUrl = mediaUrl;
        this.time = time;
    }

    public URL getWebSiteUrl() {
        return webSiteUrl;
    }

    public URL getImgUrl() {
        return imgUrl;
    }

    public URL getMediaUrl() {
        return mediaUrl;
    }

    public String getTime() {
        return time;
    }

    public boolean hasImgUrl() {
        return imgUrl != EMPTY_URL;
    }

    public boolean hasMediaUrl() {
        return mediaUrl != EMPTY_URL;
    }

    public boolean hasWebSiteUrl() {
        return webSiteUrl != EMPTY_URL;
    }

    public String info() {
        final StringBuilder sb = new StringBuilder("Item{");
        sb.append("name=").append(name);
        sb.append(", time='").append(time).append('\'');
        sb.append(", webSiteUrl=").append(webSiteUrl);
        sb.append(", imgUrl=").append(imgUrl);
        sb.append(", mediaUrl=").append(mediaUrl);
        sb.append('}');
        return sb.toString();
    }

    public void setImgUrl(URL imgUrl) {
        this.imgUrl = imgUrl;
    }
    
    public void setMediaUrl(URL mediaUrl) {
        this.mediaUrl = mediaUrl;
    }

    public void setWebSiteUrl(URL webSiteUrl) {
        this.webSiteUrl = webSiteUrl;
    }

}

File: src/main/kt/cz/skup5/jEvropa2/data/Item.kt
package cz.skup5.jEvropa2.data

import java.net.URI

/**
 * Represents an multimedia (audio or video) record.
 *
 * @author Skup5
 */
data class Item(
        /** The item unique id. */
        override val id: Int = ID_NONE,
        /** The item title/label. */
        override val name: String,
        /** The date/time value parsed from web site. */
        var timestamp: String,
        /** The absolute url of item web site. */
        var webSiteUri: URI = EMPTY_URI,
        /** The img absolute url of item. */
        var imgUri: URI = EMPTY_URI,
        /** The multimedia absolute url of item. */
        var mediaUri: URI = EMPTY_URI,
        /** The multimedia type of item. It is relative to [mediaUri].*/
        var mediaType: MultiMediaType = MultiMediaType.UNKNOWN,
        /** The show id which this item belongs to. */
        var showId: Int = ID_NONE
) : E2Data(id, name) {

    fun hasImgUri(): Boolean = imgUri !== EMPTY_URI

    fun hasMediaUri(): Boolean = mediaUri !== EMPTY_URI

    fun hasWebSiteUri(): Boolean = webSiteUri !== EMPTY_URI

    /**
     * Returns text representation of current item.
     */
    fun info(): String {
        val sb = StringBuilder(name)
        with(sb) {
            append(" (")
            append(mediaType).append(" ")
            append(timestamp)
            append(", web=").append(webSiteUri)
            append(", img=").append(imgUri)
            append(", media=").append(mediaUri)
            append(')')
        }
        return sb.toString()
    }

}
